implementations/micrometer-registry-dynatrace/src/main/java/io/micrometer/dynatrace/v2/DynatraceExporterV2.java

```diff
@@ -0,0 +1,362 @@
+                    try {
+                        throwIfValueIsInvalid(measurement.getValue());
+                        return createMetricBuilder(meter, meter.getId().getName())
+                                .setDoubleCounterValueDelta(measurement.getValue())
+                                .serialize();
+                    } catch (MetricException e) {
+                        logger.warn(String.format("Could not serialize metric with name %s: %s", meter.getId().getName(), e.getMessage()));
+                    } catch (IllegalArgumentException iae) {
+                        logger.warn(String.format("Illegal value for metric with name %s: %s Dropping...", meter.getId().getName(), iae.getMessage()));
+                    }
+                    return null;
+                })
+                .filter(Objects::nonNull);
+    }
+
+    private Metric.Builder createMetricBuilder(Meter meter, String metricKey) {
+        return metricBuilderFactory
+                .newMetricBuilder(metricKey)
+                .setDimensions(fromTags(meter.getId().getTags()))
+                .setTimestamp(Instant.ofEpochMilli(clock.wallTime()));
+    }
+
+    private String createMetricKey(String name, String tagValueRepresentation) {
+        if (name.endsWith(".percentile")) {
+            return null;
+        }
+
+        String dynatraceTag;
+        switch (tagValueRepresentation) {
+            case "total":
+                dynatraceTag = "sum";
+                break;
+            case "value":
+                dynatraceTag = "";
+                break;
+            case "percentile":
+                // drop lines that have a tag value representation of percentile.
+                // we use the 0 percentile to determine minimum values for summaries and timers
+                // so this will export 0 for every timer and summary.
+                logger.warn("dropping percentile value");
+                return null;
+            default:
+                dynatraceTag = tagValueRepresentation;
+                break;
+        }
+
+        if (dynatraceTag.isEmpty()) {
+            return name;
+        }
+        return String.format("%s.%s", name, dynatraceTag);
+    }
+
+    private DimensionList fromTags(List<Tag> tags) {
+        return DimensionList.fromCollection(
+                tags.stream()
+                        .map(x -> Dimension.create(x.getKey(), x.getValue()))
+                        .collect(Collectors.toList()));
+    }
+
+    static <T> Stream<T> streamOf(Iterable<T> iterable) {
+        return StreamSupport.stream(iterable.spliterator(), false);
+    }
+
+    private void send(List<String> metricLines) {
+        try {
+            String body = metricLines.stream().collect(Collectors.joining(System.lineSeparator()));
+            logger.debug("sending lines:\n" + body);
+
+            httpClient.post(endpoint)
+                    .withHeader("Authorization", "Api-Token " + config.apiToken())
```

Comment:
As per the docs you should specify `Content-Type: text/plain` and please also add a `User-Agent: micrometer` header to aid troubleshooting (or ideally `micrometer/1.8.0` if you can easily retrieve the version number dynamically).